neutron_lbaas_dashboard/static/app/core/openstack-service-api/barbican.service.js

```diff
@@ -82,5 +85,56 @@
       });
     }
 
+    /**
+     * @name horizon.app.core.openstack-service-api.barbican.createSecret
+     * @description
+     * Create the new secret.
+     * @param {object} spec
+     * The result is an object with property "secret_ref".
+     * @returns promise of createSecret.
+     */
+
+    function createSecret(spec) {
+      var deferred = $q.defer();
```

Comment:
$q is a global variable? here $q is not as an arguement of the function.

Reply:
as mentioned, $q is not a variable, but a 'included/imported' service.

Reply:
Thanks, do not js much after all. but the JS syntax seems weird to put an import statement into a variable.

Reply:
different syntax..